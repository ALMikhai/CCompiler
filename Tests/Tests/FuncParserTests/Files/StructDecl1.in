struct a { 
    int a, b[], *c;
};

int main() {
    struct a a1;
    return 0;
}