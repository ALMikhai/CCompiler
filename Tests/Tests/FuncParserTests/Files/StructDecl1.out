\-TRANSLATIONUNIT
  |-DECL
  | |-DECLSPEC
  | | |-STRUCTSPEC
  | | | |-a
  | | | \-STRUCTDECLLIST
  | | |   \-STRUCTDECL
  | | |     |-DECLSPEC
  | | |     | |-INT
  | | |     \-STRUCTDECLARATORLIST
  | | |       |-DECLARATOR
  | | |       | \-a
  | | |       |-DECLARATOR
  | | |       | \-ARRAY
  | | |       |   |-b
  | | |       |   \-EMPTY
  | | |       \-DECLARATOR
  | | |         |-POINTER
  | | |         \-c
  | \-INITDECLARATORLIST
  \-FUNC
    |-DECLSPEC
    | |-INT
    |-INITDECLARATOR
    | \-DECLARATOR
    |   \-FUNC
    |     |-main
    |     \-EMPTY
    \-COMPOUNDSTAT
      |-DECLLIST
      | \-DECL
      |   |-DECLSPEC
      |   | |-STRUCTSPEC
      |   | | |-a
      |   | | \-EMPTY
      |   \-INITDECLARATORLIST
      |     \-INITDECLARATOR
      |       \-DECLARATOR
      |         \-a1
      \-STATLIST
        \-RETURN
          \-0

